host/sgx/create.c

```diff
@@ -310,13 +312,26 @@ static oe_result_t _calculate_enclave_size(
                  + (size_settings->num_stack_pages * OE_PAGE_SIZE) +
                  OE_PAGE_SIZE; // guard page
 
-    /* Compute the control size in bytes (6 pages total) */
-    control_size = 6 * OE_PAGE_SIZE;
+    if (image->elf.tdata_size)
+    {
+        tls_size += oe_round_up_to_multiple(
+            image->elf.tdata_size, image->elf.tdata_align);
+    }
+    if (image->elf.tbss_size)
+    {
+        tls_size += oe_round_up_to_multiple(
+            image->elf.tbss_size, image->elf.tbss_align);
+    }
+
+    tls_size = oe_round_up_to_multiple(tls_size, OE_PAGE_SIZE);
+
+    /* Compute the control size in bytes (5 pages total) */
+    control_size = 5 * OE_PAGE_SIZE;
```

Comment:
I'm not sure whether the `5` here is related to `OE_SGX_NUM_CONTROL_PAGES+1`; if it is, I think it would be nicer to use the macro.

Reply:
Yes, it’s nicer to use macros.

sgxload.h:
OE_SGX_NUM_CONTROL_PAGES:  1 page TCS + 2 pages SSA + 1 page GAUARD
eeid.h:
OE_SGX_TCS_CONTROL_PAGES: 1 page TCS + 2 pages SSA + 1 page GAUARD + 1 page TD

Reply:
commit code: 3b14072cc1450cc9be56b86cfa38905caf43e12a

Reply:
Thanks for the fixes! I'm afraid I don't quite understand why those two are different, in my mind they should really be the same. From the places where they are used, it looks like `OE_SGX_TCS_CONTROL_PAGES == OE_SGX_NUM_CONTROL_PAGES + 1`? Now that there's `OE_SGX_NUM_CONTROL_PAGES` I think we should remove `OE_SGX_TCS_CONTROL_PAGES` from `eeid.h`.

Reply:
Remove `OE_SGX_NUM_CONTROL_PAGES,` and `#define OE_SGX_TCS_CONTROL_PAGES 4 ` is defined in openenclave/internal/sgx/sgxproperties.h, and `#include <openenclave/internal/properties.h>` in eeid.c. What do you think of it.

Reply:
I would suggest adding the definitions to include/openenclave/internal/constants_x64.h instead, under the category of _Open Enclave layout constant definitions_. 

Regarding why the values are different, the original `OE_SGX_NUM_CONTROL_PAGES` was defined to calculate how to get to the TD page offset in both it's use in `_add_control_pages` and `_patch`, whereas EEID is concerned about getting the total TCS size, which would need to include the TD page itself. 

I would suggest adding appropriate comments in the header definition to disambiguate this, for example:

```c
// SGX control pages, excluding thread data and local storage:
// 1 TCS page + 2 SSA pages + 1 guard page
#define OE_SGX_TCS_CONTROL_PAGES 4

#define OE_SGX_TCS_THREAD_DATA_PAGES 1
```

Reply:
Thank you for your comments, the code has been modified according to the above suggestions, please review again. 
2a2808148d257e127f48ee1a3d8be8730f593be8